src/azure-cli/azure/cli/command_modules/appconfig/_snapshotmodels.py

```diff
@@ -0,0 +1,147 @@
+    :ivar int items_count:
+        The number of key-values in the built snapshot.
+    :ivar dict[str, str] tags:
+        Dictionary of tags of the snapshot.
+    :ivar str items_link:
+        Endpoint that can be used to query keys-values in a snapshot
+    :ivar int retention_period:
+        Number of days for which an archived snapshot will be kept before being deleted.
+    :ivar int status_code:
+        The status code returned in the event where snapshot creation fails.
+    '''
+
+    def __init__(self,
+                 name,
+                 status,
+                 filters,
+                 etag=None,
+                 status_code=None,
+                 composition_type=None,
+                 created=None,
+                 expires=None,
+                 size=None,
+                 items_count=None,
+                 tags=None,
+                 items_link=None,
+                 retention_period=None,
+                 ):
+
+        self.name = name
+        self.status = status
+        self.filters = filters
+        self.status_code = status_code
+        self.etag = etag
+        self.composition_type = composition_type
+        self.created = created.isoformat() if isinstance(created, datetime) else str(created)
+        self.expires = expires.isoformat() if isinstance(expires, datetime) else (str(expires) if expires else None)
+        self.size = size
+        self.items_count = items_count
+        self.tags = tags
+        self.items_link = items_link
+        self.retention_period = retention_period
+
+    def __str__(self):
+        return "\netag: " + self.etag + \
+            "\nName: " + self.name + \
+            "\nStatus: " + self.status + \
+            "\nFilters: " + (str(self.filters) if self.filters else '') + \
+            "\nComposition Type: " + self.composition_type + \
+            "\nCreated: " + self.created + \
+            "\nExpires: " + self.expires + \
+            "\nSize: " + str(self.size) + \
+            "\nItem count: " + str(self.items_count) + \
+            "\nTags: " + (str(self.tags) if self.tags else '{}') + \
+            "\nItems Link: " + self.items_link + \
+            "\nRetention Period: " + str(self.retention_period)
+
+    @classmethod
+    def from_json(cls, data_dict: any):
+        return cls(
+            name=data_dict.get("name", None),
+            status=data_dict.get("status", None),
+            filters=data_dict.get("filters", None),
+            etag=data_dict.get("etag", None),
+            composition_type=data_dict.get("composition_type", None),
+            created=data_dict.get("created", None),
+            expires=data_dict.get("expires", None),
+            size=data_dict.get("size", 0),
+            items_count=data_dict.get("items_count", 0),
+            tags=data_dict.get("tags", {}),
+            items_link=data_dict.get("items_link", None),
+            retention_period=data_dict.get("retention_period", 0)
```

Comment:
I guess the default value should be null here since the value is not actually 0. The defaults shown here are  assigned to the snapshot object in the event where the returned JSON from a request does not contain a given property. Eg., when fields are specified.

Reply:
I think null is better than 0, because 0 is a valid value for retention period

Reply:
Got it, I guess same could be done for the `items_count` and `size` too.

Reply:
This has been updated, thanks!